### PR TITLE
Refactor/shellcheck bin

### PR DESCRIPTION
**Pull request by forslund**
## Description
Adds shellcheck stage to github actions as discussed in #3015 checking the shellscripts in `bin`. This should be extended to also check the _dev-setup.sh_ and _scripts_ folder etc. but this is a start.

This also fixes the issues detected by shellcheck in the checked folder, the biggest changes were in `mycroft-config`.

The current check excludes the following checks:
- SC1091: Avoids errors when shellcheck can't find sourced file
- SC2034: Unused variables, for example colors that aren't used yet
- SC2012: use of ls, from what I can see in our case this is fine (wc -l)

## How to test
Ensure github action Shellcheck runs ok.

## Contributor license agreement signed?
CLA [ x]

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -66,3 +66,14 @@ jobs:
       run: if [[ ${{ matrix.python-version }} == 3.9 ]]; then bash <(curl -s https://codecov.io/bash); fi
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Shell check mycroft tools
+      uses: ludeeus/action-shellcheck@master
```
Comment (krisgesling) [inline .github/workflows/unit_test.yml:75]: Can we set the action-shellcheck to a specific commit rather always using master?

This is a [recommended practice by Github](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_iduses) as it removes the ability for the action author or some other malicious actor to add unexpected code to that repo.

`actions/checkout@v2` is fine as it's an official Github Action, but this one I'd say is more vulnerable.

**Review by krisgesling — CHANGES_REQUESTED** [review]
Thanks Ake, this looks great. I'm going to spend a few hours using these scripts just to put them through some manual testing, but it looks right.

Just one small change request

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (forslund) [reply .github/workflows/unit_test.yml:75]: good suggestion, I'll fix that tonight

**Comment by forslund** [comment]
Manual tests are most welcome, I tried to exercise all updated scripts but may well have missed things. Especially in the more complex scripts.

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (forslund) [reply .github/workflows/unit_test.yml]: Updated locking it to the current commit on master.

**Review by krisgesling — APPROVED** [review]
Great thanks!

I did a fairly thorough exercise of all the scripts and didn't find anything out of place so good to go.

The CI is just a bit backed up from the translation updates to the Marketplace.

**Comment by PureTryOut** [comment]
> The current check excludes the following checks:
>    SC1091: Avoids errors when shellcheck can't find sourced file
>    SC2034: Unused variables, for example colors that aren't used yet
>    SC2012: use of ls, from what I can see in our case this is fine (wc -l)

Please let shellcheck ignore those errors with [a directive](https://github.com/koalaman/shellcheck/wiki/Directive) if you want to ignore those errors @forslund

**Comment by forslund** [comment]
I decided against using the directive in the script files since it in my opinion it cluttered up the files unnecessarily but I can make that change if it's preferred.

**Comment by PureTryOut** [comment]
I disagree with it making it cluttered, and right now Shellcheck still spits out warnings which is ugly and imo worrying. There are fine reasons to ignore some warnings but shellcheck should really be told to ignore them. Now people might get a "oh this warning is like one of those others we ignore, I can ignore this one too" feeling while it might this time actually valid complain. Let's be explicit that we allow these particular "mistakes".

**Comment by forslund** [comment]
Your reasoning is valid too I'll make an update to this effect

**Pull request merged**